### PR TITLE
new plug and play machines (5x NOT WORKING XaviX, 2x WORKING XaviX) + epo_epp2 redump

### PR DESCRIPTION
**Pull request by mamehaze**
new WORKING machines
------------
Let's! TV Play Keroro Sergeant Keroro Platoon Panic! It's a Big Slapstick Battle (Japan) [David Haywood, Peter Wilhelmsen, TeamEurope]
Tokyo Friend Park II Special! (Japan) [David Haywood, Peter Wilhelmsen, TeamEurope]

new NOT WORKING machines
-----------
Tottoko Hamtaro Ham Ham Big Circus! (Japan) [David Haywood, Peter Wilhelmsen, TeamEurope]
TV Mahjong Promotion Game - 4-Player Game (Japan) [David Haywood, Peter Wilhelmsen, TeamEurope]
Card Scan! Excite Stage Japan National Soccer Team (Japan) [David Haywood, TeamEurope]
Let's catch black bass! Excite Fishing (Japan) [David Haywood, TeamEurope]
Let's Hit a Home Run! Excite Stadium (Japan) [David Haywood, TeamEurope]

- replace excite ping pong 2 dump with a good one [TeamEurope]

### CODE REVIEW

**Comment by mamehaze** [comment]
@cuavas this is now all the ones with 8-bit ROM Globs that TeamEurope have and can dump

others are 16-bit ROMs, and Klaus needs a new solder station too, as his has died, so this should be it for now.

corrections to the names, as outlined above, would be appreciated, as I only have the existing translations to go on.

The reorganization of data at the top is a work in progress to see what I can strip out of the driver entirely (redundant info) without removing the log of what still needs to be dumped.  I won't be making any further changes to it at this time until some more of the 16-bit games are dumped and put where they belong (either in SuperXavix or Xavix2 as those are the platforms that require 16-bit ROMs)

**Comment by mamehaze** [comment]
ok, TeamEurope managed to dump one of the non-pads cases too

File: src/mame/mame.lst [hunk 1]
```diff
@@ -45856,15 +45857,19 @@ epo_eppk                        //
 epo_epps                        //
 epo_esdx                        //
 epo_esht                        //
+epo_stad
+epo_fish                        //
 epo_guru                        //
 epo_hamd                        //
+epo_mj                          //
 epo_mms                         //
```
Comment (cuavas) [inline src/mame/mame.lst:45865]: These are out of order again.

File: src/mame/tvgames/xavix.cpp [hunk 2]
```diff
@@ -2455,6 +2533,9 @@ CONS( 2001, epo_strk,  0,          0,  xavix,            epo_epp,  xavix_state,
 
 CONS( 2002, rad_jcon,  0,          0,  xavix,            rad_jcon, xavix_state,          init_xavix,    "Radica / Takara / SSD Company LTD",            "Play TV Jr. Construction", MACHINE_IMPERFECT_SOUND )
 
+// ホームラン打とうぜ! エキサイトスタジアム  
+CONS( 2000, epo_stad,  0,          0,  xavix,            epo_epp,  xavix_state,          init_xavix,    "Epoch / SSD Company LTD",                      "Let's Hit a Home Run! Excite Stadium (Japan)", MACHINE_NOT_WORKING | MACHINE_IMPERFECT_SOUND )
```
Comment (cuavas) [inline src/mame/tvgames/xavix.cpp:2537]: “Home Run Datouze! Excite Stadium”

File: src/mame/tvgames/xavix.cpp [hunk 3]
```diff
@@ -2482,16 +2563,22 @@ CONS( 2003, epo_epp2,   0,         0,  xavix,            epo_epp,  xavix_state,
 // 愛ちゃんに挑戦！エキサイトピンポン
 CONS( 2006, epo_epp3,   0,         0,  xavix,            epo_epp,  xavix_state,          init_xavix,    "Epoch / SSD Company LTD",                      "Ai-chan ni Chousen! Excite Ping Pong (Japan)", MACHINE_NOT_WORKING | MACHINE_IMPERFECT_SOUND )
 
-// 大モノ釣ろうぜ！ エキサイトフィッシングＤＸ 
+// TV麻雀 昇段対局～4人打ち
+CONS( 2003, epo_mj,     0,         0,  xavix_i2c_24lc02_mj,  epo_mj,   xavix_i2c_mj_state,   init_xavix,    "Epoch / SSD Company LTD",                      "TV Mahjong Promotion Game - 4-Player Game (Japan)", MACHINE_NOT_WORKING | MACHINE_IMPERFECT_SOUND )
```
Comment (cuavas) [inline src/mame/tvgames/xavix.cpp:2567]: “TV Mahjong Shoudan Taikyoku - 4-nin Uchi”

File: src/mame/tvgames/xavix.cpp [hunk 4]
```diff
@@ -2482,16 +2563,22 @@ CONS( 2003, epo_epp2,   0,         0,  xavix,            epo_epp,  xavix_state,
...
+
+// ブラックバス釣ろうぜ！ エキサイトフィッシング 
+CONS( 2001, epo_fish,  0,          0,  xavix,            xavix,    xavix_state,          init_xavix,    "Epoch / SSD Company LTD",                      "Let's catch black bass! Excite Fishing (Japan)", MACHINE_NOT_WORKING | MACHINE_IMPERFECT_SOUND )
```
Comment (cuavas) [inline src/mame/tvgames/xavix.cpp:2570]: “Black Bass Tsurouze! Excite Fishing”

File: src/mame/tvgames/xavix.cpp [hunk 5]
```diff
@@ -2500,6 +2587,9 @@ CONS( 2002, epo_mms,   0,          0,  xavix_i2c_24c02,  epo_mms,  xavix_i2c_sta
 // ミニモニ。ステージ！ダンスだぴょん！ぷらすっ
 CONS( 2003, epo_mmsp,  0,          0,  xavix_i2c_24c02,  epo_mms,  xavix_i2c_state,      init_xavix,    "Epoch / SSD Company LTD",                      "mini-moni Stage! Dance Dapyon! Plus (Japan)",  MACHINE_IMPERFECT_SOUND )
 
+// Let's!TVプレイ ケロロ軍曹 ケロロ小隊大パニック！ドタバタ大決戦であります
+CONS( 2006, ban_krrj,  0,          0,  xavix,            ban_krrj, xavix_state,          init_xavix,    "Bandai / SSD Company LTD",                      "Let's! TV Play Keroro Sergeant Keroro Platoon Panic! It's a Big Slapstick Battle (Japan)",  MACHINE_IMPERFECT_SOUND )
```
Comment (cuavas) [inline src/mame/tvgames/xavix.cpp:2591]: After looking at the front of the box, I don’t think “ケロロ軍曹” is part of the title.

“Let's! TV Play Keroro Shoutai Dai Panic! Dotabata Daikessen de Arimasu”

File: src/mame/tvgames/xavix_2000.cpp [hunk 6]
```diff
@@ -350,10 +366,17 @@ CONS( 2002, epo_bowl, 0, 0, xavix2000_i2c_24c04, epo_bowl,    xavix_i2c_state,
 // needs timer irq hack to boot, fails to draw main menu properly (buggy xavix2000 opcodes?)  (2002 date on PCB, 2003 ingame)
 CONS( 2003, epo_golf, 0,       0, xavix2000_i2c_24c04, ttv_lotr,   xavix_i2c_lotr_state, init_epo_mini, "Epoch / SSD Company LTD",       "Super Shot! Excite Golf (Japan)", MACHINE_NOT_WORKING | MACHINE_IMPERFECT_SOUND )
 
+// とっとこハム太郎 ハムハム大サーカス！
+CONS( 2002, epo_hamc,  0,      0, xavix2000,           epo_hamc,   xavix_epo_hamc_state, init_xavix,    "Epoch / SSD Company LTD",       "Tottoko Hamtaro Ham Ham Big Circus! (Japan)", MACHINE_NOT_WORKING | MACHINE_IMPERFECT_SOUND )
```
Comment (cuavas) [inline src/mame/tvgames/xavix_2000.cpp:370]: “Tottoko Hamtaro - Ham Ham Dai Circus!”

File: src/mame/tvgames/xavix_2000.cpp [hunk 7]
```diff
@@ -350,10 +366,17 @@ CONS( 2002, epo_bowl, 0, 0, xavix2000_i2c_24c04, epo_bowl,    xavix_i2c_state,
...
+
 // ミニモニ。パーティ！リズムでぴょん！
 // needs timer irq hack to boot
 CONS( 2003, epo_mini, 0,       0, xavix2000_i2c_24c08, ttv_lotr,   xavix_i2c_lotr_state, init_epo_mini, "Epoch / SSD Company LTD",        "mini-moni Party! Rhythm de Pyon! (Japan)", MACHINE_NOT_WORKING | MACHINE_IMPERFECT_SOUND )
 
+// カードスキャン!　エキサイトステージ サッカー日本代表チーム 
+CONS( 2006, epo_es2j,   0,     0,  xavix2000,          xavix,      xavix_state,          init_xavix,    "Epoch / SSD Company LTD",        "Card Scan! Excite Stage Japan National Soccer Team (Japan)", MACHINE_NOT_WORKING | MACHINE_IMPERFECT_SOUND )
```
Comment (cuavas) [inline src/mame/tvgames/xavix_2000.cpp:377]: “Card Scan! Excite Stage Soccer Nippon Daihyou Team”

**Comment by mamehaze** [comment]
(the current build failure is nothing to do with these changes)

File: src/mame/tvgames/xavix_m.cpp [hunk 8]
```diff
@@ -521,6 +521,18 @@ void xavix_i2c_ltv_tam_state::write_io1(uint8_t data, uint8_t direction)
...
+	if (direction & 0x02)
+	{
+		m_i2cmem->write_sda((data & 0x02) >> 1);
+	}
+
+	if (direction & 0x01)
+	{
+		m_i2cmem->write_scl((data & 0x01) >> 0);
+	}
+}
```
Comment (cuavas) [inline src/mame/tvgames/xavix_m.cpp:535]: You should also release these when the output isn’t being driven, as there will be a pull-up resistor on each of the lines:
```c++
	m_i2cmem->write_sda(BIT(data | ~direction, 1));
	m_i2cmem->write_scl(BIT(data | ~direction, 0));
```

**Comment by mamehaze** [comment]
fixed, away for the weekend now, so if anything else needs correcting please push to the branch

**Comment by cuavas** [comment]
> fixed, away for the weekend now, so if anything else needs correcting please push to the branch

Thanks, it looks OK for now.

**Pull request merged**